src/c++/library/common.cc

```diff
@@ -110,9 +110,10 @@ InferenceServerClient::UpdateInferStat(const RequestTimers& timer)
 Error
 InferInput::Create(
     InferInput** infer_input, const std::string& name,
-    const std::vector<int64_t>& dims, const std::string& datatype)
+    const std::vector<int64_t>& dims, const std::string& datatype,
+    const bool binary_input_data)
```

Comment:
Because of the way the c++ client is written, this seems very complex to do.

I have spent some time reflecting on this, and I simply _cannot_ think of an easy way to hide this detail from the gRPC client.

1. I cannot subclass InferInput because the base constructor is private.
2. I cannot store the content type of each input at the creation of the client, because inputs are not known at that point, they're only known when the inference function is called.
3. I cannot extend the inference function to take an extra parameter indicating the content type of each input because the inference function is protocol-independent, thus coming back to the issue of gRPC being aware of this detail.

Of course each of these potential solutions can be forced to work, but involve changing more and more of the client.

If you have a suggestion of a way to hide this detail from the gRPC client, I'm open to it. But for now, I cannot think of a good solution without a major refactor of the code, so I'm inclined to leave the detail "visible" (not really because it's a parameter with a default value) to the gRPC client.

Reply:
It looks like it is not possible in the current way that the API is designed. All the clients would use `triton::client::InferInput`. It seems like we don't have any other option than exposing this to both gRPC and HTTP. 

CC @tanmayv25 if he has any ideas. Otherwise, I think this should be fine.

Reply:
Discussed offline. We can add a "protocol-independent" options parameter that can be subclassed as either http infer input/requestedoutput options and grpc infer input/requestedoutput options, with the grpc one having no actual options and the http one allowing json or binary inputs/outputs.

Reply:
Approving for now. Hopefully this can be addressed in a follow up PR.